src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,97 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private Object[] array;
+    private int size;
+
+    public ArrayList() {
+        array = new Object[CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (array.length == size) {
+            resize();
+        }
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
```

Comment:
let's make two separate checks: for index and for capacity

the check for index is slightly different from your checkIndex() method - index may be equal to size